test/spec/fixtures/schema.sql

```diff
@@ -3103,6 +3114,136 @@ create table test.subscriptions(
   primary key(subscriber, subscribed)
 );
 
+-- Data representations feature
+DROP DOMAIN IF EXISTS public.color CASCADE;
+CREATE DOMAIN public.color AS INTEGER CHECK (VALUE >= 0 AND VALUE <= 16777215);
+
+CREATE OR REPLACE FUNCTION color(json) RETURNS public.color AS $$
+  SELECT color($1 #>> '{}');
+$$ LANGUAGE SQL IMMUTABLE;
```

Comment:
Hm, could you elaborate on why this works? The `$1 #>> '{}` is puzzling to me. Since the array is empty(`'{}'`) I don't get why it could just not be `$1::text` .

Reply:
Ah, it's because that form strips the quotes right?

```sql
select '"foobar"'::json#>>'{}';
 ?column?
----------
 foobar
```

While casting to text:

```sql
select '"foobar"'::text;
   text
----------
 "foobar"
```

Do you know if there's another way to do the same that's more obvious?

Reply:
Ah yeah, it's not the most obvious thing is it? So you're right, what's happening here is that we expect a JSON string, `"#abcde"`, and we want the actual string value, `#abcde`, so we can forward to our regular textual parser. It's not sufficient to just `$1::text` it and regex out the `"`, I think, since we need to invoke the JSON decoder in case there are JSON specific escape sequences or whatever in the string.

[From the docs](https://www.postgresql.org/docs/11/functions-json.html) we see that we can extract the text content of some given path in a JSON document with the `#>>` operator.  The empty path `{}` gives us the root element which is what we want.

One alternative syntax is to explicitly use the `json_extract_path_text` function like,

```
SELECT jsonb_extract_path_text($1, '{}');
```

Would that be more clear you think?

Reply:
Got it, all clear now!

> Would that be more clear you think?

Hm, unfortunately that function doesn't work the same as the operator it seems:
```sql
select json_extract_path_text('"foobar"'::json, '{}');
 json_extract_path_text
------------------------

```

So it seems fine to leave as it is.

Reply:
Ah, just got it to work with:

```sql
select json_extract_path_text('"foobar"'::json, VARIADIC ARRAY[]::text[]);
 json_extract_path_text
------------------------
 foobar

```
Much more verbose and not clearer at all.